### PR TITLE
Update Cluster configuration with the haproxy helper settings

### PR DESCRIPTION
**Pull request by yenienserrano**
### Description
Update Cluster configuration with the haproxy helper settings
 
### Issues Resolved
- https://github.com/wazuh/wazuh-dashboard/issues/175

### Evidence

image

### Test

You can try the imposter or a manager of this branch: [epic-20887-migrate-haproxy-helper](https://github.com/wazuh/wazuh/tree/epic-20887-migrate-haproxy-helper)

If you use a manager on that branch you will have to run this so that the manager doesn't hang. And then restart wazuh
`/var/ossec/framework/python/bin/pip3 install httpx`

Tests:

Always in `Server Management/Settings/Cluster`

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Without the `haproxy_handler` configuration, only the main settings should be rendered. | :black_circle: | :black_circle: | :black_circle: |
| With the basic `haproxy_handler` configuration, everything should render with the default values | :black_circle: | :black_circle: | :black_circle: |
| By adding the other settings, the changed settings should render. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Without the `haproxy_handler` configuration, only the main settings should be rendered.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With the basic `haproxy_handler` configuration, everything should render with the default values</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: By adding the other settings, the changed settings should render.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

These configurations go inside `<cluster></cluster>`

Basic settings:

```xml
    <haproxy_helper>
        <haproxy_disabled>yes</haproxy_disabled>
        <haproxy_address>wazuh-proxy</haproxy_address>
        <haproxy_user>haproxy</haproxy_user>
        <haproxy_password>haproxy</haproxy_password>
    </haproxy_helper>
```

All settings:

```xml
   <haproxy_helper>
        <haproxy_disabled>no</haproxy_disabled>
        <haproxy_address>wazuh-proxy</haproxy_address>
        <haproxy_user>haproxy</haproxy_user>
        <haproxy_password>haproxy</haproxy_password>
        <haproxy_port>5555</haproxy_port>
        <haproxy_protocol>http</haproxy_protocol>
        <haproxy_backend>wazuh_cluster</haproxy_backend>
        <frequency>60</frequency>
        <agent_chunk_size>100</agent_chunk_size>
        <agent_reconnection_time>10</agent_reconnection_time>
        <agent_reconnection_stability_time>60</agent_reconnection_stability_time>
        <imbalance_tolerance>0.1</imbalance_tolerance>
        <remove_disconnected_node_after>1</remove_disconnected_node_after>
        <excluded_nodes>master,worker1</excluded_nodes>
    </haproxy_helper>
```

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by JuanGarriuz** [comment]
## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Without the `haproxy_handler` configuration, only the main settings should be rendered. | :black_circle: | :black_circle: | :black_circle: |
| With the basic `haproxy_handler` configuration, everything should render with the default values | :black_circle: | :black_circle: | :black_circle: |
| By adding the other settings, the changed settings should render. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Without the `haproxy_handler` configuration, only the main settings should be rendered.</summary>

Chrome - :black_circle:

image


Firefox - :black_circle:


image

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With the basic `haproxy_handler` configuration, everything should render with the default values</summary>

Chrome - :black_circle:

image

Firefox - :black_circle:

image


Safari - :black_circle:

</details>

<details>
<summary>:black_circle: By adding the other settings, the changed settings should render.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

These configurations go inside `<cluster></cluster>`

**Review by JuanGarriuz — DISMISSED** [review]
LGTM!!!

File: plugins/main/public/controllers/management/components/management/configuration/cluster/cluster.js [hunk 1]
```diff
@@ -58,10 +85,22 @@ class WzCluster extends Component {
...
     };
+
+    if (currentConfig['com-cluster'].haproxy_helper) {
+      mainSettingsConfig = {
+        ...mainSettingsConfig,
+        haproxy_helper: {
+          ...currentConfig['com-cluster'].haproxy_helper,
+          haproxy_disabled:
+            currentConfig['com-cluster'].haproxy_helper.haproxy_disabled ===
+            true
```
Comment (Desvelao) [inline plugins/main/public/controllers/management/components/management/configuration/cluster/cluster.js:98]: **nitpick**: is the `haproxy_disabled` coming as a boolean? It seems to be configured as a string with `yes`/`no`

File: plugins/main/public/controllers/management/components/management/configuration/cluster/cluster.js [hunk 2]
```diff
@@ -58,10 +85,22 @@ class WzCluster extends Component {
     let mainSettingsConfig = {
       ...currentConfig['com-cluster'],
       disabled:
-        currentConfig['com-cluster'].disabled === 'yes'
-          ? 'disabled'
-          : 'enabled',
+        currentConfig['com-cluster'].disabled === true ? 'disabled' : 'enabled',
```
Comment (Desvelao) [inline plugins/main/public/controllers/management/components/management/configuration/cluster/cluster.js:88]: **question**: did it change the value of the `disabled` property to a boolean?

**Comment by asteriscos** [comment]
## UI

| Test | Chrome | Firefox | 
| --- |  --- | --- | 
| Without the `haproxy_handler` configuration, only the main settings should be rendered. | :black_circle: | :black_circle: | 
| With the basic `haproxy_handler` configuration, everything should render with the default values | :black_circle: | :black_circle: |
| By adding the other settings, the changed settings should render. | :black_circle: | :black_circle: | 

**Details**
<details>
<summary>:black_circle: Without the `haproxy_handler` configuration, only the main settings should be rendered.</summary>

Chrome - :black_circle:




Firefox - :black_circle:





</details>

<details>
<summary>:black_circle: With the basic `haproxy_handler` configuration, everything should render with the default values</summary>

Chrome - :black_circle:



Firefox - :black_circle:






</details>

<details>
<summary>:black_circle: By adding the other settings, the changed settings should render.</summary>

Chrome - :black_circle:



Firefox - :black_circle:






</details>

These configurations go inside `<cluster></cluster>`

File: plugins/main/public/controllers/management/components/management/configuration/cluster/cluster.js (hunk 1 above)
Comment (Desvelao) [reply plugins/main/public/controllers/management/components/management/configuration/cluster/cluster.js:98]: It seems in 4.7.1 is a boolean, so this fixes an unnoticed bug.

File: plugins/main/public/controllers/management/components/management/configuration/cluster/cluster.js (hunk 2 above)
Comment (yenienserrano) [reply plugins/main/public/controllers/management/components/management/configuration/cluster/cluster.js:88]: It seems to come from previous versions returning a boolean

image

**Review by asteriscos — APPROVED** [review]

**Review by asteriscos — APPROVED** [review]

**Pull request merged**